src/components/CommentsVoteManager.jsx

```diff
@@ -0,0 +1,64 @@
+import { useState } from "react";
+import { patchCommentVotes } from "../../api";
+import { ChevronUpIcon, ChevronDownIcon } from "@heroicons/react/16/solid";
+
+export default function CommentsVoteManager({ comment_id, initialVotes }){
+  const [voteModifier, setVoteModifier] = useState(0);
+  const [error, setError] = useState(null);
+  const [isVoting, setIsVoting] = useState(false);
+
+  const toggleUpvote = () => {
+    const newModifier = voteModifier === 1 ? 0 : 1;
+    const voteChange = newModifier - voteModifier;
+
+    setIsVoting(true);
+    setError(null);
+
```

Comment:
I think this component is a perfect case use for optimistic rendering and would strongly recommend optimistically rendering a patched vote